File: longest-common-prefix/LongestCommonPrefix.cs
public class Solution
{
    public string LongestCommonPrefix(string[] strs)
    {
        var result = strs[0];
        for (int i = 1; i < strs.Length; i++)
        {
            var minLength = Math.Min(result.Length, strs[i].Length);
            var length = 0;
            for (int j = 0; j < minLength; j++)
                if (result[j] == strs[i][j])
                    length++;
                else
                    break;

            if (length == 0)
                return "";
            result = result.Substring(0, length);
        }
        return result;
    }
}

File: longest-common-prefix/longestCommonPrefix.go
func longestCommonPrefix(strs []string) string {
    var min = func(a,b int) int {
    if(a < b) { return a }
    return b
}
    result := strs[0]
    for i:= 1; i < len(strs); i++ {
        minlen := min(len(result), len(strs[i]))
        result = result[:minlen]
        for j:= 0; j < minlen; j++ {
            if(result[j] != strs[i][j]) {
                result = result[:j]
                break
            }
        }
    }

    return result
}
